src/MediatR/Mediator.cs

```diff
@@ -21,17 +21,32 @@ public class Mediator : IMediator
         private readonly ConcurrentDictionary<Type, Type> _genericHandlerCache;
         private readonly ConcurrentDictionary<Type, Type> _wrapperHandlerCache;
 
+        private PublishAsyncOptions _publishOption;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Mediator"/> class.
         /// </summary>
         /// <param name="singleInstanceFactory">The single instance factory.</param>
         /// <param name="multiInstanceFactory">The multi instance factory.</param>
-        public Mediator(SingleInstanceFactory singleInstanceFactory, MultiInstanceFactory multiInstanceFactory)
+        public Mediator(SingleInstanceFactory singleInstanceFactory, MultiInstanceFactory multiInstanceFactory) :

```

Comment:
Adding one more ctor is a breaking chance since some DI containers (such as Simple Injector) will stop working. So I'd keep only one and rely on the options parameters sent to an overload accepting it.